### ISSUE TITLE
Support topNSigma sampling in `generate`

### ISSUE THREAD

**Issue by pramodith** [huggingface/transformers#39850]
### Feature request

topNSigma is a sampling technique that guarantees that outputs are agnostic to temperature, while eliminating unrealistic tokens thereby reducing the chances of an illegitimate generation.

It's implementation centers around finding the std in the logits space and retaining the tokens whose logit scores are within the top-n standard deviations from the token corresponding to the max logit.

### Motivation

Their sampling technique shows improvements on a number of benchmarks when used with the Llama-3 family of models.

>  Extensive experiments across reasoning
and creative writing tasks demonstrate that our
method consistently outperforms existing approaches, with particularly significant improvements in high-temperature settings.

Link to the paper: https://aclanthology.org/2025.acl-long.528.pdf

### Your contribution

I'd love to create a PR for this!

**Comment by zucchini-nlp** [comment]
Hey @pramodith ! Great, I think this can be added as a new [decoding method with remote code](https://huggingface.co/docs/transformers/generation_strategies#custom-decoding-methods). Hosting decoding methods on the hub allows us to bypass a long review cycle and let the users start experimenting easily. If the decoding method becomes very popular, we will add it to core library as well

In case you need any help with custom remote decoding methods, ping @(gante)

**Comment by pramodith** [comment]
Thanks @zucchini-nlp, that's pretty cool didn't know HF had support for custom decoding techniques using remote code! I've implemented the method in this HF repo https://huggingface.co/Pramodith/topN_sigma_generation. Feel free to close out this issue.

**Comment by zucchini-nlp** [comment]
Amazing, I will close the issue then. If you want the repo to be transferred to https://huggingface.co/transformers-community org, please let us know ;)

**Comment by pramodith** [comment]
Oh sure happy for that to be transferred, let me know if I need to do anything on my end.

**Comment by zucchini-nlp** [comment]
cc @gante for transferring it
